src/bundle/Controller/RecommendationController.php

```diff
@@ -84,29 +74,26 @@ public function showRecommendationsAction(Request $request): Response
             $this->recommendationService->sendDeliveryFeedback($request->get(BasicRecommendationRequest::OUTPUT_TYPE_ID_KEY));
         }
 
-        return $this->templateEngine->renderResponse($template, [
+        return $response->setContent(
+            $this->getTwig()->render($template, [
             'recommendations' => $event->getRecommendationItems(),
             'templateId' => Uuid::uuid4()->toString(),
-            ],
-            $response
+            ])
         );
     }
 
-    /**
-     * @param bool $value
-     */
     public function sendDeliveryFeedback(bool $value): void
     {
         $this->sendDeliveryFeedback = $value;
     }
 
-    /**
-     * @param string|null $template
-     *
-     * @return string
-     */
+    protected function getTwig(): Environment
```

Comment:
could be changed to: ([ref](https://symfony.com/doc/current/service_container/service_subscribers_locators.html))
```
protected function twig(): Environment
{
    return $this->container->get(__METHOD__);
}
```